providers/vsphere/resources/resourceclient/host.go

```diff
@@ -140,6 +145,49 @@ func HostLicenses(client *vim25.Client, hostID string) ([]types.LicenseManagerLi
 	return res, nil
 }
 
+// GetHostTags retrieves tags for a host system using the vSphere vAPI
+// If the vAPI is not available, it will return an empty array instead of an error.
+// This maintains backward compatibility with vSphere environments that don't use tags.
+func (c *Client) GetHostTags(ctx context.Context, hostRef types.ManagedObjectReference, conf *inventory.Config) []string {
+	// Create vAPI REST client
+	restClient := rest.NewClient(c.Client.Client)
+
+	// Get credentials from connection config
+	creds, err := vault.GetPassword(conf.Credentials)
+	if err != nil {
+		return []string{}
+	}
+
+	userInfo := url.UserPassword(creds.User, string(creds.Secret))
+	err = restClient.Login(ctx, userInfo)
+	if err != nil {
+		return []string{}
+	}
+
+	tagManager := tags.NewManager(restClient)
+
+	// Get attached tags for the host
+	attachedTags, err := tagManager.GetAttachedTags(ctx, hostRef)
+	if err != nil {
+		return []string{}
+	}
+
+	// Convert tags to string format: "category:tag"
+	tagStrings := make([]string, len(attachedTags))
+	for i, tag := range attachedTags {
+		// Get category information
+		category, err := tagManager.GetCategory(ctx, tag.CategoryID)
+		if err != nil {
+			// If we can't get category, just use tag name
+			tagStrings[i] = tag.Name
+			continue
+		}
+		tagStrings[i] = fmt.Sprintf("%s:%s", category.Name, tag.Name)
+	}
+
+	return tagStrings
+}
```

Comment:
This looks to me the same as `GetVmTags`. Is it possible to have the main logic of these two extracted into one func?